Backend/internal/auth/login.go

```diff
@@ -132,7 +140,7 @@ func (u *User) checkPassword() error {
 	// i.e. even if the user puts ' or 1= '1
 	// the hashed value will not be an injection command
 
-	matches := CredentialsMatch(u.Email, hashedPassword)
+	matches := _db.CredentialsMatch(u.Email, hashedPassword)
 	if matches == 0 {
 		return errors.New("invalid credentials")
 	} else if matches > 1 { // if more than 1 result is returned
```

Comment:
one thing we could try is that maybe we could add a constraint to the schema? so that we never get into a situtation of duplicate entries? something like:
```sql
CONSTRAINT no_dupes UNIQUE (email, password)
```

Reply:
oooo i see
ill look into that